### ISSUE TITLE
Generic Parceler @FragmentArg does not compile

### ISSUE THREAD

**Issue by WonderCsabo** [androidannotations/androidannotations#1950]
This is a template, please adjust it accordingly to your specific issue.<br/>
You may also want to read our [issue guide](https://github.com/androidannotations/androidannotations/wiki/HowToWriteAnIssue).

**AndroidAnnotations version**: 4.2.0

**Android compile SDK version**:  24

**Annotated code**:
```java
public class MyFragment<T extends MyParcel> extends Fragment {

    @FragmentArg
    T answer;
    
}
```

Expected:
This should be valid, because the upper bound of the type parameter is indeed a Parcel bean, and its subtypes should be Parcel beans as well. 

Actual: AA adds a validation error, because it things it is not Parcel bean.

**Comment by WonderCsabo** [comment]
@johncarl81 this is correct that in this case we should consider `T` as Parcel bean, right?

**Comment by johncarl81** [comment]
@WonderCsabo, this is a little tricky.  Parceler requires the bean in use to be annotated with `@Parcel`.  Extending a bean that is annotated is not enough.  I assume your end Fragment would have a concrete implementation to base your generated code on, would checking if a bean passes as a `Parceler` bean at that point be appropriate?  In that case, I'd suggest looking at the WIP I have here https://github.com/johncarl81/transfuse/commit/99330429ef15517480bff6c5db32ce694e6acb0e that can be used to resolve generic types to their concrete implementation declaration.

**Comment by WonderCsabo** [comment]
@johncarl81 thanks for your quick answer, and sorry for my delayed one. Unfortunately in my case, the generic field is annotated, where we do not yet know the actual subclasses. If there is no guarantee if a superclass is a Parcel, then all of its subclasses are parcels as well by some type of contract, i guess we cannot validate this. Maybe Parceler should enforce this in some way?

**Comment by johncarl81** [comment]
Correct me if I'm wrong, but in your example, there would be a concrete implementation that AA would use to generate the extension class, something like this:
```java
@EFragment
public class ToUseFragment extends MyFragment<RealParcel> {...}
```
where `RealParcel` would be an `@Parcel` annotated bean.

If this is the case we can validate when generating the extension class for `ToUseFragment`.
